### PR TITLE
chore: Implement list Pop/Erase functionality with QList

### PR DESCRIPTION
**Pull request by romange**
Adjusted OpRem/OpBPop/OpPop

### CODE REVIEW

File: src/core/qlist.cc [hunk 1]
```diff
@@ -355,6 +355,40 @@ void QList::Push(string_view value, Where where) {
   }
 }
 
+string QList::Pop(Where where) {
```
Comment (BorysTheDev) [inline src/core/qlist.cc:358]: Just to think: we can create our own string_view type (that can contain string_view or short string for 19 symbols) and return it without allocations

File: src/server/list_family.cc [hunk 2]
```diff
@@ -572,46 +606,77 @@ OpResult<uint32_t> OpRem(const OpArgs& op_args, string_view key, string_view ele
...
+      count = -count;
+      where = QList::TAIL;
+    }
+
+    auto it = ql->GetIterator(where);
+    auto is_match = [&](const QList::Entry& entry) {
+      if (is_int) {
+        return entry.is_int() && entry.ival() == ival;
+      }
+      return entry == elem;
```
Comment (BorysTheDev) [inline src/server/list_family.cc:631]: return (is_int && entry.is_int() && entry.ival() == ival) || entry == elem

**Review by BorysTheDev — APPROVED** [review]

File: src/core/qlist.cc (hunk 1 above)
Comment (dranikpg) [reply src/core/qlist.cc:358]: You don't need to optimize by space, so it can be just a variant<string, string_view>

Comment (romange) [reply src/core/qlist.cc:358]: we can not return string_view if we pop. in any case it's not urgent and can not be done easily during the transition.
there is no regression here.

File: src/server/list_family.cc (hunk 2 above)
Comment (romange) [reply src/server/list_family.cc:631]: will fix in the subsequent pr

**Pull request merged**